src/univariate/continuous/ksdist.jl

```diff
@@ -16,52 +16,52 @@ function cdf(d::KSDist,x::Float64)
     n = d.n
     b = x*n
     # known exact values
-    if b <= 0.5
-        return 0.0
-    elseif b <= 1.0
+    if b <= 1/2
+        return 0
```

Comment:
...but `exp(lfact(n)+n*(log(2*b-1)-log(n)))` below will be `Float64` so the return type depends on the value of `b`.